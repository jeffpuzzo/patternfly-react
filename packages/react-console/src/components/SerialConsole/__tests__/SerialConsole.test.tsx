import React from 'react';

import { render, screen } from '@testing-library/react';

import { SerialConsole } from '../SerialConsole';
import { constants } from '../../common/constants';

const { CONNECTED, DISCONNECTED, LOADING } = constants;

describe('SerialConsole', () => {
  beforeAll(() => {
    window.HTMLCanvasElement.prototype.getContext = () => ({ canvas: {} } as any);
  });

  test('in the LOADING state', () => {
    render(
      <div data-testid="test-id">
        <SerialConsole
          onData={jest.fn()}
          onConnect={jest.fn()}
          onDisconnect={jest.fn()}
          status={LOADING}
          textLoading="My text for Loading"
        />
      </div>
    );
    expect(screen.getByTestId('test-id').innerHTML).toMatchSnapshot();
  });

  test('in the DISCONNECTED state', () => {
    render(
      <div data-testid="test-id">
        {' '}
        <SerialConsole
          onData={jest.fn()}
          onConnect={jest.fn()}
          onDisconnect={jest.fn()}
          status={DISCONNECTED}
          textDisconnected="My text for Disconnected"
          textConnect="My text for Connect"
        />
      </div>
    );
    expect(screen.getByTestId('test-id').innerHTML).toMatchSnapshot();
  });

  describe('with CONNECTED state', () => {
    beforeAll(() => {
      window.HTMLCanvasElement.prototype.getContext = () =>
        ({ canvas: {}, createLinearGradient: jest.fn(), fillRect: jest.fn() } as any);
      global.window.matchMedia = () => ({
        addListener: jest.fn(),
        removeListener: jest.fn(),
        matches: true,
        media: undefined,
        onchange: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn()
      });
    });

    test('renders', () => {
      render(
        <div data-testid="test-id">
          <SerialConsole
            cols={33}
            rows={44}
            status={CONNECTED}
            onData={jest.fn()}
            onConnect={jest.fn()}
            onDisconnect={jest.fn()}
            textDisconnect="My text for Disconnect"
          />
        </div>
      );

      expect(screen.getByTestId('test-id').innerHTML).toMatchSnapshot();
    });
  });
});
