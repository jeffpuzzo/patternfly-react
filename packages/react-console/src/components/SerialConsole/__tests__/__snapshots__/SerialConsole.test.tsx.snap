// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerialConsole in the DISCONNECTED state 1`] = `" <div class=\\"pf-c-console__serial\\"><div class=\\"pf-c-empty-state\\"><div class=\\"pf-c-empty-state__content\\"><div class=\\"pf-c-empty-state__body\\">My text for Disconnected</div><button aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-1\\">My text for Connect</button></div></div></div>"`;

exports[`SerialConsole in the LOADING state 1`] = `"<div class=\\"pf-c-console__actions-serial\\"><button aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-secondary\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-secondary-1\\">Disconnect</button><button aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-secondary\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-secondary-2\\">Reset</button></div><div class=\\"pf-c-console__serial\\"><div class=\\"pf-c-empty-state\\"><div class=\\"pf-c-empty-state__content\\"><div class=\\"pf-c-empty-state__icon\\"><span class=\\"pf-c-spinner pf-m-xl\\" role=\\"progressbar\\" aria-valuetext=\\"Loading...\\" aria-label=\\"Contents\\"><span class=\\"pf-c-spinner__clipper\\"></span><span class=\\"pf-c-spinner__lead-ball\\"></span><span class=\\"pf-c-spinner__tail-ball\\"></span></span></div><div class=\\"pf-c-empty-state__body\\">My text for Loading</div></div></div></div>"`;

exports[`SerialConsole with CONNECTED state renders 1`] = `"<div class=\\"pf-c-console__actions-serial\\"><button aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-secondary\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-secondary-3\\">My text for Disconnect</button><button aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-secondary\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-secondary-4\\">Reset</button></div><div class=\\"pf-c-console__serial\\"><div class=\\"pf-c-console__xterm\\" role=\\"list\\"><div dir=\\"ltr\\" class=\\"terminal xterm focus\\" tabindex=\\"0\\"><div class=\\"xterm-accessibility\\"><div class=\\"xterm-accessibility-tree\\" role=\\"document\\"><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div><div role=\\"listitem\\" tabindex=\\"-1\\" style=\\"height: 0px;\\"></div></div><div class=\\"live-region\\" aria-live=\\"assertive\\"></div></div><div class=\\"xterm-viewport\\" style=\\"background-color: rgb(0, 0, 0);\\"><div class=\\"xterm-scroll-area\\"></div></div><div class=\\"xterm-screen\\"><div class=\\"xterm-helpers\\"><textarea class=\\"xterm-helper-textarea\\" aria-label=\\"Terminal input\\" aria-multiline=\\"false\\" autocorrect=\\"off\\" autocapitalize=\\"off\\" spellcheck=\\"false\\" tabindex=\\"0\\"></textarea><span class=\\"xterm-char-measure-element\\" aria-hidden=\\"true\\">W</span><div class=\\"composition-view\\"></div></div><canvas class=\\"xterm-text-layer\\" style=\\"z-index: 0;\\"></canvas><canvas class=\\"xterm-selection-layer\\" style=\\"z-index: 1;\\"></canvas><canvas class=\\"xterm-link-layer\\" style=\\"z-index: 2;\\"></canvas><canvas class=\\"xterm-cursor-layer\\" style=\\"z-index: 3;\\"></canvas></div></div></div></div>"`;
